frontend/src/components/pages/FridgeManagement/FridgeCheckIns/index.tsx

```diff
@@ -0,0 +1,290 @@
+      }
+
+      set(newEndDate, {
+        year: endDateState.getFullYear(),
+        month: endDateState.getMonth(),
+        date: endDateState.getDate(),
+      });
+
+      if (endTime) {
+        set(newEndDate, {
+          hours: endTime.getHours(),
+          minutes: endTime.getMinutes(),
+        });
+      }
+
+      setCheckInForm({
+        target: { name: "endDate", value: newEndDate.toString() },
+      });
+    } else {
+      setCheckInForm({
+        target: { name: "endDate", value: "" },
+      });
+    }
+    setFormErrors({
+      ...formErrors,
+      dateRange: "",
+    });
+  };
+  const handleChange = (
+    e:
+      | React.ChangeEvent<HTMLInputElement>
+      | React.ChangeEvent<HTMLTextAreaElement>,
+    name: string,
+  ) => {
+    if (name === "startDate") {
+      const time = new Date(`11/11/1970 ${e.target.value}`);
+      setStartTime(time);
+      const newtime = set(new Date(time), {
+        year: dateRange[0].year,
+        month: dateRange[0].month.number - 1,
+        date: dateRange[0].day,
+      });
+
+      setCheckInForm({ target: { name: "startDate", value: newtime } });
+      setFormErrors({
+        ...formErrors,
+        timeRange: "",
+      });
+    } else if (name === "endDate") {
+      const time = new Date(`11/11/1970 ${e.target.value}`);
+      setEndTime(time);
+
+      if (dateRange[1]) {
+        const newtime = set(new Date(time), {
+          year: dateRange[1].year,
+          month: dateRange[1].month.number - 1,
+        });
+        // NOTE: handles case for if end date is midnight, BUT what about HH:mm pm startTime and HH:mm am endTime?
+        if (e.target.value === "00:00") {
+          newtime.setDate(dateRange[1].day + 1);
+          setEndTime(set(new Date(endTime!), { date: endTime!.getDate() + 1 }));
+        } else {
+          newtime.setDate(dateRange[1].day);
+        }
+        setCheckInForm({ target: { name: "endDate", value: newtime } });
+      }
+
+      setFormErrors({
+        ...formErrors,
+        timeRange: "",
+      });
+    } else {
+      // notes
+      setCheckInForm({ target: { name: "notes", value: e.target.value } });
+    }
+  };
+
+  const validateForm = () => {
+    const newErrors = {
+      timeRange: "",
+      dateRange: "",
+    };
+    let valid = true;
+    if (!checkInFormValues.startDate || !checkInFormValues.endDate) {
+      valid = false;
+      newErrors.dateRange = ErrorMessages.bothDateFieldsRequired;
+    }
+    if (!startTime || !endTime) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.bothTimeFieldsRequired;
+    } else if (isAfter(startTime!, endTime!)) {
+      valid = false;
+      newErrors.timeRange = ErrorMessages.endTimeBeforeStartTime;
+    }
+    setFormErrors(newErrors);
+    return valid;
+  };
+
+  const onSaveClick = async () => {
+    const isValid = validateForm();
+    if (!isValid) {
+      return;
+    }
+    const res = await CheckInAPIClient.createCheckIn(checkInFormValues);
```

Comment:
I've tried testing with entering April 10 to April 14 and I'm getting this as the `checkInFormValues` that it's sending in the request:
```javascript
endDate: "Mon Apr 04 2022 21:47:14 GMT-0400 (Eastern Daylight Time)"
isAdmin: false
notes: "apr 10-15"
startDate: "Mon Apr 04 2022 12:29:00 GMT-0400 (Eastern Daylight Time)"
```
it's taking the current date no matter what range we enter